src/app/registration/equal-to-validator/equal-to.directive.ts

```diff
@@ -0,0 +1,40 @@
+import { Directive, Input, OnChanges } from '@angular/core';
+import { Validator, NG_VALIDATORS, AbstractControl, FormControl } from '@angular/forms';
+import { IEqualToResult } from './equal-to.interfaces';
+
+@Directive({
+    selector: '[shEqualTo]',
+    providers: [{provide: NG_VALIDATORS, useExisting: EqualToDirective, multi: true}]
+})
+
+export class EqualToDirective implements Validator, OnChanges {
```

Comment:
What you mean under 'such complexity'? It's just an easiest custom validator according to Angular 2 docs :). The only complexity we have to add here was onChange hook, because we want to cover the case when after password_confirm was set and user change password field again we still validate password confirmation field.